### ISSUE TITLE
 RFC: Add (optional?) entry for commit hash to `mods.json`

### ISSUE THREAD

**Issue by GeckoEidechse** [R2Northstar/VerifiedMods#6]
Related to #5 

Having the commit hash next to version and checksum would be useful to detect someone shifting around the tag that was used validate a previous version which could allow hiding commits if we only look at differential changes.

This would change

```json
{
    "Fifty.mp_frostbite": {
        "DependencyPrefix": "Fifty-Frostbite",
        "Versions" : [
            { 
                "Version": "0.0.1",
                "Checksum": "8cf111c9ac2ab1521677769702e10be4db8d47c0eca16bda197aceb895cf2b5c"
            }
        ]
    }
}
```
to

```json
{
    "Fifty.mp_frostbite": {
        "DependencyPrefix": "Fifty-Frostbite",
        "Versions" : [
            { 
                "Version": "0.0.1",
                "CommitHash": "446f6e2774206c6f6f6b20626568696e6420746865206375727461696e212121",
                "Checksum": "8cf111c9ac2ab1521677769702e10be4db8d47c0eca16bda197aceb895cf2b5c"
            }
        ]
    }
}
```

I'm not sure if we should make the key optional or not as maps might have an issue in general with being hosted on GitHub due to their file sizes.

**Comment by Alystrasz** [comment]
Map size can not be an issue at all; @uniboi showed it's possible to split the map in small chunks (uploadable to GitHub), that are regrouped together when uploading to Thunderstore.

Example repo: https://github.com/uniboi/s2space
Associated mod: https://northstar.thunderstore.io/package/odds/s2space/

**Comment by Alystrasz** [comment]
Otherwise, including commit hashes seems a good idea to me.

**Comment by GeckoEidechse** [comment]
Closed by #11
